reconcile/aus/base.py

```diff
@@ -394,6 +431,14 @@ def fetch_current_state(
             for upgrade_policy in upgrade_policies:
                 upgrade_policy["cluster"] = spec.cluster
                 current_state.append(ControlPlaneUpgradePolicy(**upgrade_policy))
+            for node_pool in get_node_pools(ocm_api, spec.cluster.id):
+                node_upgrade_policies = get_node_pool_upgrade_policies(
+                    ocm_api, spec.cluster.id, node_pool["id"]
+                )
+                for upgrade_policy in node_upgrade_policies:
+                    upgrade_policy["cluster"] = spec.cluster
+                    upgrade_policy["node_pool"] = node_pool["id"]
+                    current_state.append(NodePoolUpgradePolicy(**upgrade_policy))
```

Comment:
* can multiple nodepool upgrades exist at the same time for a cluster?
* can a controlplane upgrade exist at the same time as nodepoolupgrades? (e.g. controlplane upgrade to X+1 while nodepools upgrade to X)

i'm asking this because i think there are some places in our code that assume there is only one AbstractUpgradePolicy object per cluster. if multiple can exist we need to revisite (not a huge problem i guess)

Reply:
The diff method sets the mutex for node pool upgrades. Thus our code will not create an upgrade policy if an upgrade was scheduled already. 


Will test the following just for understanding:
- create multiple node pool upgrade policies
- schedule control plane upgrade and schedule a node pool upgrade

Reply:
A cluster can have both, a control plane upgrade and a node pool upgrade scheduled

Reply:
A node pool can only have one upgrade policy: 
```
  "reason": "Upgrade policy with upgrade type 'NodePool' already exists. Only a single policy of the same type is supported",
```

Reply:
do i understand correctly that our code will make sure that only one thing upgrades for a cluster at a single point in time? so either the control plane, or nodepool 1, or nodepool 2, ... ?

Reply:
Indeed, node pool policies rely on the same mutex logic as control plane upgrades